Documentation Update (#40)
nf-core/nanoseq: docs/output.md

* [GraphMap](#graphmap) - mapping for long reads
* [MiniMap2](#minimap2) - mapping for long reads
* [SortBam](#sortbam) - coordinate sort BAM files using SAMtools

> * [bedtools](#bedtools) - Create bigWig and bigBed files 

Reviewer: ```suggestion
* [bedtools](#bedtools) - create bigWig and bigBed files 
```

[Guppy](https://nanoporetech.com/nanopore-sequencing-data-analysis)

*Description*:  
Guppy will demultiplex and barcode the data given from an ONT device. The flowcell, kit and barcode kit must be given in the command line if demultiplexing needed. This step can by bypassed using the `--skip_demultiplexing` parameter when initiating the pipeline. The output folders will be separated into the barcodes from the kit used and unclassified. The output in each barcode folder is then merged into one fastq file for easier downstream processing.

*Output directories*:

* `guppy/basecalling/barcode*/`  
  FastQ files output for each barcode
* `guppy/basecalling/unclassified/`  
  FastQ files output that are unclassified

> * `guppy/fastq`

Reviewer 2: ```suggestion
* `guppy/fastq/`
```

[Guppy](https://nanoporetech.com/nanopore-sequencing-data-analysis)

*Description*:  

> Guppy will demultiplex and barcode the data given from an ONT device. The flowcell, kit and barcode kit must be given in the command line if demultiplexing needed. This step can by bypassed using the `--skip_demultiplexing` parameter when initiating the pipeline. The output folders will be separated into the barcodes from the kit used and unclassified. The output in each barcode folder is then merged into one fastq file for easier downstream processesing. 

Reviewer: ```suggestion
Guppy will demultiplex and barcode the data given from an ONT device. The flowcell, kit and barcode kit must be given in the command line if demultiplexing needed. This step can by bypassed using the `--skip_demultiplexing` parameter when initiating the pipeline. The output folders will be separated into the barcodes from the kit used and unclassified. The output in each barcode folder is then merged into one fastq file for easier downstream processing.
```

*Output directories*:

* `guppy/basecalling/barcode*/`  
  FastQ files output for each barcode
* `guppy/basecalling/unclassified/`  
  FastQ files output that are unclassified
* `guppy/fastq`
  Merged output of fastq files into one fastq for each barcode

> ## Sequncing Quality Control

Reviewer 3: Typo?
-> Sequencing

> ## Sequncing Quality Control

Reviewer: ```suggestion
## Sequencing Quality Control
```

*Documentation*:  
[PycoQC](https://github.com/a-slide/pycoQC), [NanoPlot](https://github.com/wdecoster/NanoPlot)

*Description*:  
PycoQC and NanoPlot give general quality metrics about the sequencing run. It provides information about the distribution of read length, read length over time, number of reads per barcode and other general stats.
![PycoQC - Number of Reads per Barcode plot](images/NumberofReadsperBarcode.png)

*Output directories*:

* `pycoQC/`  
  An .html file output is produced that includes a run summary and graphical representation of distribution of read length, distribution of read quality scores, mean read quality per sequence length, output per channel over experiment time, output over experiment time, read quality over experiment time, readlength over experiment time, and percentage of reads per barcode.
* `nanoplot/summary/`  

>   An output of a .png file of a statistical run summary and an html summary file of overall run.

Reviewer: What do you mean by `statistical run summary` and `overall run`?

Author: Just to differentiate it between the individual fastqc Nanoplot report from the overall Nanoplot report. I will reword this to be more clear.

Reviewer: That would be perfect, because I honestly had no idea what you were talking about.
:confused:

> 

Reviewer: ```suggestion
```

The files resulting from the alignment with graphmap2 or minimap2 of individual libraries are not saved by default so this directory will not be present in your results. You can override this behaviour with the use of the `--save_align_intermeds` flag in which case it will contain the coordinate sorted alignment files in [`*.bam`](https://samtools.github.io/hts-specs/SAMv1.pdf) format.

![ALIGNER - Alignment per barcode](images/mqc_samtools_alignment_plot_1.png)

*Output directories*:

> * `graphmap/`  

Reviewer 2: ```suggestion
* `graphmap2/`  
```

>   If the `--aligner graphmap` parameter is used, the sorted and indexed bam files will be output here.

Reviewer 2: ```suggestion
  If the `--aligner graphmap2` parameter is used, the sorted and indexed bam files will be output here.
```

* `minimap2/`  
  If the `--aligner minimap2` parameter is used, the sorted and indexed bam files will be output here.
* `<ALIGNER>/samtools_stats/`  
  `*.flagstat`, `*.idxstats` and `*.stats` files generated from the alignment files using SAMtools.

## bigWig and bigBed 

*Documentation*:
[`BEDTools`](https://github.com/arq5x/bedtools2/), [`bedGraphToBigWig`](http://hgdownload.soe.ucsc.edu/admin/exe/), [`bedToBigBed`](http://hgdownload.soe.ucsc.edu/admin/exe/)

*Description*:
Creation of bigWig and bigBed coverage tracks for visualisation. This can be bypassed by setting the parameters `--skip_bigwig` and/or `--skip_bigbed`.

*Output directories*:

> * `<ALIGNER>/bigwig`  

Reviewer 2: ```suggestion
* `<ALIGNER>/bigwig/`  
```

  The bigWig files will be output here.

> * `<ALIGNER>/bigbed`  

Reviewer 2: ```suggestion
* `<ALIGNER>/bigbed/`  
```

The files resulting from the alignment with graphmap or minimap2 of individual libraries are not saved by default so this directory will not be present in your results. You can override this behaviour with the use of the `--save_align_intermeds` flag in which case it will contain the coordinate sorted alignment files in [`*.bam`](https://samtools.github.io/hts-specs/SAMv1.pdf) format.

![ALIGNER - Alignment per barcode](images/mqc_samtools_alignment_plot_1.png)

*Output directories*:

* `graphmap/`  
  If the `--aligner graphmap` parameter is used, the sorted and indexed bam files will be output here.
* `minimap2/`  
  If the `--aligner minimap2` parameter is used, the sorted and indexed bam files will be output here.
* `<ALIGNER>/samtools_stats/`  
  `*.flagstat`, `*.idxstats` and `*.stats` files generated from the alignment files using SAMtools.

## bigWig and bigBed 

*Documentation*:
[`BEDTools`](https://github.com/arq5x/bedtools2/), [`bedGraphToBigWig`](http://hgdownload.soe.ucsc.edu/admin/exe/), [`bedToBigBed`](http://hgdownload.soe.ucsc.edu/admin/exe/)

*Description*:

> Creation of bigWig and bigBed coverage tracks for visualisation. This can by bypassed by setting the parameters `--skip_bigwig` and/or `--skip_bigbed`.

Reviewer: ```suggestion
Creation of bigWig and bigBed coverage tracks for visualisation. This can be bypassed by setting the parameters `--skip_bigwig` and/or `--skip_bigbed`.
```

* `multiqc/Project_multiqc_report.html`  
  MultiQC report - a standalone HTML file that can be viewed in your web browser
* `multiqc/multiqc_data/`  
  Directory containing parsed statistics from the different tools used in the pipeline
* `multiqc/multiqc_plots/`
  Directory containing the image files of the graphs included in MultiQC

For more information about how to use MultiQC reports, see [http://multiqc.info](http://multiqc.info)

## Other results

**Pipeline information**

  *Documentation*:  
  [Nextflow!](https://www.nextflow.io/docs/latest/tracing.html)

  *Description*:  
  Nextflow provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to trouble-shoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage.

  *Output directories*:
  * `pipeline_info/`  
    * Reports generated by the pipeline - `pipeline_report.html`, `pipeline_report.txt` and `software_versions.csv`.
    * Reports generated by Nextflow - `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `pipeline_dag.svg`.

>     * Reformatted design files used as input to the pipeline - `design_reads.csv` and `design_controls.csv`.

Reviewer 2: ```suggestion
    * Reformatted samplesheet files used as input to the pipeline - `samplesheet_reformat.csv`.
```

* `multiqc/Project_multiqc_report.html`  
  MultiQC report - a standalone HTML file that can be viewed in your web browser
* `multiqc/multiqc_data/`  
  Directory containing parsed statistics from the different tools used in the pipeline
* `multiqc/multiqc_plots/`

>   Directory containing the image files of the graphs included in MultiQC

Reviewer 2: We should also include a description of the `pipeline_info` files generated by Nextflow. See https://github.com/nf-core/chipseq/blob/master/docs/output.md#other-results
